src/main/java/org/mule/extension/sftp/internal/util/UriUtils.java

```diff
@@ -0,0 +1,280 @@
+   * @param filePath the path to the file.
+   * @return a {@link URI} representing the resolved Path between the two given arguments.
+   */
+  public static URI createUri(String basePath, String filePath) {
+    String fullPath;
+    try {
+      if (filePath.length() > 0) {
+        if (Boolean.TRUE.equals(isAbsolute(filePath))) {
+          fullPath = filePath;
+        } else {
+          fullPath = addSeparator(basePath) + filePath;
+        }
+      } else {
+        fullPath = removeSeparator(basePath);
+      }
+
+      if (fullPath.split("\\r?\\n").length != 1) {
+        throw new IllegalPathException("Path contains newline character: " + fullPath);
+      }
+
+      return new URI(null, null, fullPath, null);
+    } catch (URISyntaxException e) {
+      throw new IllegalPathException("Cannot convert given path into a valid Uri", e);
+    }
+  }
+
+  /**
+   * Adds a separator at the end of the given path. If the path already ends with the separator, then this method does nothing.
+   */
+  private static String addSeparator(String path) {
+    return (path.endsWith(SEPARATOR) || path.length() == 1) ? path : path + SEPARATOR;
+  }
+
+  /**
+   * Removes the separator at the end of the given path. If the path does not end with the separator, then this method does
+   * nothing.
+   */
+  private static String removeSeparator(String path) {
+    return (!path.endsWith(SEPARATOR) || path.length() == 1) ? path : path.substring(0, path.length() - 1);
+  }
+
+  /**
+   * @param path the path to check
+   * @return a {@link Boolean} indicating if the path is absolute or not.
+   */
+  private static Boolean isAbsolute(String path) {
+    return (path.length() > 0 && path.startsWith(SEPARATOR));
+  }
+
+  /**
+   * @param uri the uri to normalize.
+   * @return a new normalized uri normalized according to {@link URI#normalize()}.
+   */
+  public static URI normalizeUri(URI uri) {
+    URI normalizedUri = uri.normalize();
+    String path = normalizedUri.getPath();
+    if (path.endsWith(SEPARATOR)) {
+      try {
+        normalizedUri = new URI(null, null, path.substring(0, path.length() - 1), null).normalize();
+      } catch (URISyntaxException e) {
+        throw new MuleRuntimeException(e);
+      }
+    }
+    return normalizedUri;
+  }
+
+  /**
+   * Creates a new uri by trimming the last path fragment from the given uri. Can also be thought as getting the 'parent' uri.
+   *
+   * @param uri the uri to trim.
+   * @return the trimmed or 'parent' uri.
+   */
+  public static URI trimLastFragment(URI uri) {
+    Integer index = uri.getPath().lastIndexOf(SEPARATOR);
+    return index != -1 ? createUri(uri.getPath().substring(0, index)) : null;
+  }
+
+  /**
+   * Creates a regex pattern from the given glob expression.
+   *
+   * @param globPattern glob pattern to be converted to a regular expression
+   * @return regular expression
+   * @throws PatternSyntaxException
+   */
+  public static String toRegexPattern(String globPattern) {
```

Comment:
It would be practically impossible to find a bug in this unending switch/case. At this level of complexity, this should be split into a tokenizer, syntax parser and validator and a collector logic to gather the result, at the very least.

Reply:
It doesn't seem to be tested

Reply:
Indeed, it is evident that the complexity appears to be higher than the optimal level. The inclusion of this class occurred while integrating the file-commons project into the SFTP project. As for the tests, there exists a considerable number of test cases evaluating this method within [this](https://github.com/mulesoft/mule-sftp-connector/tree/feature/W-12049455/src/test/java/org/mule/extension/file/common/api/matcher) directory. The sequence of execution is as follows: PathMatcherPredicate -> getPredicateForFilename -> predicateType.getPredicate -> UriUtils.toRegexPattern.